src/odemis/driver/smaract.py

```diff
@@ -4913,6 +4922,406 @@ def _cancelReference(self, future):
+            logging.debug("Finished referencing.")
+
+            # We could save the referencing parameters to memory here. This could be useful for the validation,
+            # because it initializes the system by loading the stored referencing parameters. However, we don't
+            # currently use validation, so there is no need to store the values every time we reference.
+        except si.Error as error:
+            logging.debug(self._getReadableSIError(error))
+            # This occurs if a stop command interrupts referencing
+            if error.code == si.ErrorCode.CANCELED:
+                logging.info("Referencing stopped: %s", error)
+                raise CancelledError()
+            elif future._must_stop.is_set():
+                raise CancelledError()
+            else:
+                logging.error("Referencing failed: %s", error)
+                raise
+        except CancelledError:
+            logging.debug("Referencing cancelled.")
+            raise  # No fuss, pass it as-is
+        except Exception:
+            logging.exception("Referencing failure.")
+            raise
+        finally:
+            # Update the referenced values for the channels
+            for ch, num in self._channels.items():
+                self.referenced._value[ch] = bool(si.GetProperty_i32(self.id, si.EPK(ps.Property.CH_IS_VALID, num, 0)))
+            # If the measuring system is stable and ready to produce accurate results start the polling
+            if bool(si.GetProperty_i32(self.id, si.EPK(ps.Property.SYS_IS_STABLE, 0, 0))):
+                # Start polling thread
+                self._polling_thread = util.RepeatingTimer(1, self._updatePosition, "Position polling")
+                self._polling_thread.start()
+                self._updatePosition()
+            else:
+                raise ValueError("The system did not become stable after the referencing procedure.")
+            # We only notify after updating the position so that when a listener
+            # receives updates both values are already updated.
+            self.referenced.notify(self.referenced.value)
+
+            # The adjustment state should be "disabled" after automatic adjustment. If the referencing
+            # procedure was stopped prematurely, right after it was set to "manual", it might be in the
+            # wrong state --> make sure it's set to disabled here.
+            if si.GetProperty_i32(self.id, si.EPK(ps.Property.AF_ADJUSTMENT_STATE, 0, 0)) != ps.AdjustmentState.DISABLED:
+                si.SetProperty_i32(self.id, si.EPK(ps.Property.AF_ADJUSTMENT_STATE, 0, 0), ps.AdjustmentState.DISABLED)
+                self._wait_for_progress_event(ps.AdjustmentState.DISABLED, timeout=600)  # state will be DISABLED when done
+            si.SetProperty_i32(self.id, si.Property.EVENT_NOTIFICATION_ENABLED << 16 | ps.EventType.AF_ADJUSTMENT_PROGRESS, si.DISABLED)
+
+    def _getReadableSIError(self, error):
```

Comment:
The code returns either a SI or a PS error name, so I'd suggest to rename it `getReadableError`, or maybe `getErrorName`.